src/argilla/client/feedback/training/base.py

```diff
@@ -0,0 +1,292 @@
+#  Copyright 2021-present, the Recognai S.L. team.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import os
+import textwrap
+from abc import ABC, abstractmethod
+from typing import TYPE_CHECKING, List, Optional, Union
+
+import argilla as rg
+from argilla.client.feedback.training.schemas import (
+    TrainingTaskMappingForTextClassification,
+)
+from argilla.client.models import Framework
+from argilla.training import ArgillaTrainer as ArgillaTrainerV1
+
+os.environ["PYTORCH_ENABLE_MPS_FALLBACK"] = "1"
+
+if TYPE_CHECKING:
+    import spacy
+
+
+class ArgillaTrainer(ArgillaTrainerV1):
+    def __init__(
+        self,
+        dataset: "rg.FeedbackDataset",
+        training_task_mapping: TrainingTaskMappingForTextClassification,
```

Comment:
As we're in the `ArgillaTrainer` already, maybe it makes more sense to just name this `task_mapping`?

```suggestion
        task_mapping: TrainingTaskMappingForTextClassification,
```